wezterm-gui/src/termwindow/mouseevent.rs

```diff
@@ -843,6 +843,10 @@ impl super::TermWindow {
             MouseCursor::Text
         }));
 
+        if let Some(pane) = self.get_active_pane_or_overlay() {
+            context.set_cursor(Some(pane.get_mouse_cursor_shape()));
+        }
```

Comment:
Not sure if this is the right way to get the correct pane to get the mouse cursor shape.

I think I also need to correctly include the hyperlink hover code in here, but figured I'd keep it simple for now.

As an additional note, I only added handling for OSC 22 in the `mouse_event_terminal` handler since it's the only one that seemed relevant. The OSC 22 control sequence should be emitted by a running terminal app and the other events seem related to native WezTerm events?

Do any of the other event handlers need to take into account the OSC 22 control sequence?

Reply:
GH won't let me suggest into the text above your change, so I'm sketching it out here. What I think you want is something like this:

```rust
        context.set_cursor(Some(if self.current_highlight.is_some() {
            // When hovering over a hyperlink, show an appropriate
            // mouse cursor to give the cue that it is clickable
            MouseCursor::Hand
        } else if outside_window {
            MouseCursor::Arrow
        } else if let Some(shape) = pane.get_mouse_cursor_shape() {
             // Apply the mapping here. This mapping logic should
             // be factored out into a helper function that you call here
              match shape.as_str() {
               "text" => MouseCursor::Text,
               ..
               _ => MouseCursor::Text,
            } 
        } else if pane.is_mouse_grabbed() {
            MouseCursor::Arrow
        } else {
            MouseCursor::Text
        }));
```

Reply:
Addressed in https://github.com/wez/wezterm/pull/6292/commits/b67cf84b159515d2789c946019763bb3dca61ed6 and https://github.com/wez/wezterm/pull/6292/commits/683d1348a35a558dc781faa135a4de71f8cea49c